xds/src/main/java/io/grpc/xds/orca/OrcaMetricReportingServerInterceptor.java

```diff
@@ -61,6 +65,10 @@ public static OrcaMetricReportingServerInterceptor getInstance() {
     return INSTANCE;
   }
 
+  public static void setMetricRecorder(MetricRecorder mr) {
```

Comment:
There is no reason for this to be static. I'll be commenting on that in the gRFC. This should just be a reference passed when creating the interceptor.

Reply:
In latest commit, updated `setMetricRecorder`  be non-static and made the field `metricRecorder` non-static and volatile for thread safety and updated corresponding unit tests.

Reply:
The field should be `final`. There's no need to change the reference. Just pass the instance to the constructor.

Reply:
Still is a set method rather than part of the constructor.  Since MetricRecorder is mutable, there is no reason to change this after initialization.